rc/extra/go-tools.kak

```diff
@@ -45,17 +45,83 @@ def go-disable-autocomplete -docstring "Disable gocode completion" %{
     unalias window complete go-complete
 }
 
+decl -hidden str go_format_tmp_dir
+
+def go-format -docstring "Custom formatter for go files" %{
+    %sh{
+        dir=$(mktemp -d -t kak-go.XXXXXXXX)
+        printf %s\\n "set buffer go_format_tmp_dir ${dir}"
+        printf %s\\n "eval -no-hooks write ${dir}/buf"
+    }
+    %sh{
+        dir=${kak_opt_go_format_tmp_dir}
+        goimports -e -srcdir ${kak_buffile} -w ${dir}/buf 2> ${dir}/stderr
+        if [ $? ]; then
+            cp ${dir}/buf ${kak_buffile}
```

Comment:
It's a path, so it can contain anything. Usually we assume that it won't contain double quotes (otherwise we would end up escaping things everywhere).